Source/JavaScriptCore/assembler/MacroAssemblerX86_64.h

```diff
@@ -2472,16 +2472,53 @@ class MacroAssemblerX86_64 : public MacroAssemblerX86Common {
 
     void vectorSplat(SIMDLane lane, RegisterID src, FPRegisterID dest)
     {
-        ASSERT(scalarTypeIsIntegral(lane));
-        UNUSED_PARAM(lane);
-        UNUSED_PARAM(src);
-        UNUSED_PARAM(dest);
+        m_assembler.movq_rr(src, dest);
+        switch (lane) {
+        case SIMDLane::i64x2:
+            m_assembler.movddup(dest, dest);
+            return;
+        case SIMDLane::i32x4:
+            m_assembler.shufps(dest, dest, 0);
+            return;
+        case SIMDLane::i16x8:
+            m_assembler.pshuflw(dest, dest, 0);
+            m_assembler.punpcklqdq(dest, dest);
+            return;
+        case SIMDLane::i8x16:
+            vectorReplaceLane(SIMDLane::i8x16, TrustedImm32(1), src, dest);
+            m_assembler.pshuflw(dest, dest, 0);
+            m_assembler.punpcklqdq(dest, dest);
+            return;
+        default:
+            RELEASE_ASSERT_NOT_REACHED();
+        }
+    }
+
+    void vectorSplat(SIMDLane lane, FPRegisterID src, FPRegisterID dest)
+    {
+        switch (lane) {
+        case SIMDLane::f32x4:
+            if (src != dest)
+                m_assembler.pshufd(dest, src, 0);
+            else
+                m_assembler.shufps(dest, dest, 0);
+            return;
+        case SIMDLane::f64x2:
+            if (src != dest)
+                m_assembler.movapd_rr(src, dest);
```

Comment:
Do we need this? I think we can always just do `m_assembler.movddup(dest, src);`